File: 2주차/programmers2_3.py
from collections import deque

def solution(priorities, location):
    answer = 0
    max_pri = max(priorities)
    prior_dq = deque()
    num_dq = deque()
    
    for i, p in enumerate(priorities):
        prior_dq.append(p)
        num_dq.append(i)
        
        
    while True:
        max_pri = max(prior_dq)
        p = prior_dq.popleft()
        n = num_dq.popleft()
        if p == max_pri:
            answer += 1
            if n == location :
                break
        else:
            prior_dq.append(p)
            num_dq.append(n) 

    return answer

File: 2주차/ain/programmers2_3.java
import java.util.Collections;
import java.util.PriorityQueue;

public class programmers2_3 {

    static class Solution {
        public static int solution(int[] priorities, int location) {
            int answer = 0;

            // 우선순위 큐 생성
            PriorityQueue<Integer> priorityQueue = new PriorityQueue<>(Collections.reverseOrder());

            // 큐 세팅
            for(int i : priorities) {
                priorityQueue.add(i);
            }

            while(!priorityQueue.isEmpty()) {
                for(int i = 0; i < priorities.length; i++) {
                    if (priorityQueue.peek() == priorities[i]) {
                        answer++;
                        priorityQueue.poll();
                        if (i == location) return answer;
                    }
                }
            }

            return answer;
        }
    }

    public static void main(String[] args) {
        int[] priorities = {1, 1, 9, 1, 1, 1};
        int location = 0;
        System.out.println(Solution.solution(priorities, location));
    }
}
